### ISSUE TITLE
Client Javadoc warnings

### ISSUE THREAD

**Issue by fpj** [pravega/pravega#3373]
**Problem description**
Running `./gradlew clean javadocs` produces a lot of warnings due to violations in the client javadocs.

**Problem location**
Client javadocs.

**Suggestions for an improvement**
Fix the javadoc violations.

**Comment by fpj** [comment]
The output:

```
$ ./gradlew clean javadocs

> Task :javadocs 
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/Checkpoint.java:32: warning: no @return
    ByteBuffer toBytes();
               ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/EventStreamWriter.java:64: error: reference not found
     * Note that transactions can only be open for {@link EventWriterConfig#getTransactionTimeoutTime()}.
                                                          ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/EventStreamWriter.java:68: warning - Tag @link: can't find getTransactionTimeoutTime() in io.pravega.client.stream.EventWriterConfig
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/Position.java:32: warning: no @return
    ByteBuffer toBytes();
               ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/ReaderGroup.java:157: error: reference not found
     * The configuration {@link ReaderGroupConfig#getGroupRefreshTimeMillis()} decides
                                ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/ReaderGroup.java:174: error: reference not found
     * {@link ReaderGroupConfig#getGroupRefreshTimeMillis()} which decides the duration by
              ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/ReaderGroup.java:186: warning - Tag @link: can't find getGroupRefreshTimeMillis() in io.pravega.client.stream.ReaderGroupConfig
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/ReaderGroup.java:186: warning - Tag @link: can't find getGroupRefreshTimeMillis() in io.pravega.client.stream.ReaderGroupConfig
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/StreamCut.java:67: warning: no @return
    ByteBuffer toBytes();
               ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/admin/StreamInfo.java:19: error: tag not allowed here: <li>
 *  <li> - scope of stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/admin/StreamInfo.java:20: error: tag not allowed here: <li>
 *  <li> - name of stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/admin/StreamInfo.java:21: error: tag not allowed here: <li>
 *  <li> - {@link StreamCut} which represents the current TAIL of the stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/admin/StreamInfo.java:22: error: tag not allowed here: <li>
 *  <li> - {@link StreamCut} which represents the current HEAD of the stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:33: warning: no @return
    public abstract long getOffset();
                         ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:44: error: bad use of '>'
     * Returns the number of bytes that can be read without blocking. If the number returned is > 0
                                                                                                ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:45: error: unterminated signature
     * then a call to {@link #read(byte[]))} will return data from memory without blocking. If the
                      ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:46: error: unterminated signature
     * number returned is 0 then {@link #read(byte[]))} will block. If -1 is returned this indicates
                                 ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:59: warning: no @return
    public abstract long fetchTailOffset();
                         ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:64: error: no tag name after @
     * {@see InputStream#read()}. 
       ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:73: error: no tag name after @
     * {@see InputStream#read(byte[]))}. 
       ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:99: warning: no @return
    public abstract int read(ByteBuffer dst) throws IOException;
                        ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:52: warning - Tag @link: malformed: "#read(byte[]))"
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:52: warning - Tag @link: malformed: "#read(byte[]))"
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamReader.java:76: warning - Tag @see: malformed: "InputStream#read(byte[]))"
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/byteStream/ByteStreamWriter.java:98: warning: no @return
    public abstract long fetchTailOffset();
                         ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:115: error: unknown tag: String
     * List<String> updated = stateSynchronizer.updateState((state, updates) -> {
           ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:115: error: bad use of '>'
     * List<String> updated = stateSynchronizer.updateState((state, updates) -> {
                                                                              ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:116: error: unknown tag: MyUpdate
     *      List<MyUpdate> toAdd = findUpdatesForState(state);
                ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:118: error: bad use of '>'
     *      return toAdd.stream().map(a -> a.getKey()).collect(Collectors.toList());
                                         ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:58: error: bad use of '>'
     * stateSynchronizer.updateState((state, updates) -> {
                                                       ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/state/StateSynchronizer.java:73: error: bad use of '>'
     * boolean updated = stateSynchronizer.updateState((state, updates) -> {
                                                                         ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/batch/StreamInfo.java:19: error: tag not allowed here: <li>
 *  <li> - scope of stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/batch/StreamInfo.java:20: error: tag not allowed here: <li>
 *  <li> - name of stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/batch/StreamInfo.java:21: error: tag not allowed here: <li>
 *  <li> - {@link StreamCut} which represents the current TAIL of the stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/batch/StreamInfo.java:22: error: tag not allowed here: <li>
 *  <li> - {@link StreamCut} which represents the current HEAD of the stream.
    ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/stream/notifications/notifier/AbstractNotifier.java:21: warning: no description for @param
 * @param <T>
   ^
/Users/junquf/code/pravega/client/src/main/java/io/pravega/client/ClientConfig.java:58: error: unterminated inline tag
     * If the flag {@link #isEnableTls is set, this flag decides whether to enable host name validation or not.
                                       ^
23 errors
14 warnings
```

**Comment by medvedevigorek** [comment]
Here is corresponding Lombok issue for no `@return` and no `@param` warnings that should address it once and for all - https://github.com/rzwitserloot/lombok/issues/1520 

As a temporary solution one can use `@returns` and `@params` annotations available since Lombok 1.12
